src/puppetlabs/trapperkeeper/logging.clj

```diff
@@ -72,25 +72,26 @@
             (.toInt level))
        (.setLevel root level)))))
 
-(defn configure-logger-via-file!
-  "Reconfigures the current logger based on the supplied configuration
-  file."
-  [logging-conf-file]
-  {:pre [(string? logging-conf-file)]}
+(defn configure-logger!
+  "Reconfigures the current logger based on the supplied configuration."
+  [logging-conf]
+  {:pre [(#{String java.io.File java.net.URL java.io.InputStream org.xml.sax.InputSource} (type logging-conf))]}
```

Comment:
@rlinehan good point
@KevinCorcoran yes, I think it makes sense to drop the precondition entirely as well. I only submitted a pull request to expand the list because the original code just had `string?`, which just shouldn't be.